ui/src/shared/actions/flags.ts

```diff
@@ -1,19 +1,33 @@
+import {Dispatch} from 'redux'
+import {getFlags as getFlagsRequest} from 'src/client'
+import {FlagMap} from 'src/shared/reducers/flags'
+import {RemoteDataState} from 'src/types'
 export const SET_FEATURE_FLAGS = 'SET_FEATURE_FLAGS'
+export const RESET_FEATURE_FLAGS = 'RESET_FEATURE_FLAGS'
 export const CLEAR_FEATURE_FLAG_OVERRIDES = 'CLEAR_FEATURE_FLAG_OVERRIDES'
 export const SET_FEATURE_FLAG_OVERRIDE = 'SET_FEATURE_FLAG_OVERRIDE'
 
 export type Actions =
   | ReturnType<typeof setFlags>
+  | ReturnType<typeof reset>
   | ReturnType<typeof clearOverrides>
   | ReturnType<typeof setOverride>
 
 // NOTE: this doesnt have a type as it will be determined
 // by the backend at a later time and keeping the format
 // open for transformations in a bit
-export const setFlags = flags =>
+export const setFlags = (status: RemoteDataState, flags?: FlagMap) =>
   ({
     type: SET_FEATURE_FLAGS,
-    payload: flags,
+    payload: {
+      status,
+      flags,
+    },
+  } as const)
```

Comment:
out of curiosity, what's the benefit of keeping this as a `const` rather than a custom `type`. Genuinely curious on your take

Reply:
keeps the action naming pattern cleaner. we experimented with a few when watts was doing the resource normalization work (like where you define an action and an actiontype), this had the least amount of boilerplate.  100% ergonomics and maintainability over "correctness" in testing. We found that the interface mapping / typescript benefits come more with the "typeof" during consumption than the tight coupling with the reducer (which exists as convention).